src/System.Runtime.Extensions/src/System/IO/Path.Windows.cs

```diff
@@ -70,30 +70,45 @@ private static bool IsDirectoryOrVolumeSeparator(char c)
 
         // Gets the length of the root DirectoryInfo or whatever DirectoryInfo markers
         // are specified for the first part of the DirectoryInfo name.
-        // 
         private static int GetRootLength(string path)
         {
             CheckInvalidPathChars(path);
 
             int i = 0;
             int length = path.Length;
+            int volumeSeparator = 2;
+            int uncRootLength = 2;
 
-            if (length >= 1 && IsDirectorySeparator(path[0]))
+            bool extendedSyntax = path.StartsWith(LongPathPrefix, StringComparison.Ordinal);
+            bool extendedUncSyntax = extendedSyntax && path.StartsWith(UNCLongPathPrefix, StringComparison.Ordinal);
+            if (extendedSyntax)
+            {
+                if (extendedUncSyntax)
+                {
+                    uncRootLength = UNCLongPathPrefix.Length;
+                }
+                else
+                {
+                    volumeSeparator += LongPathPrefix.Length;
+                }
+            }
+
+            if ((!extendedSyntax || extendedUncSyntax) && length >= 1 && IsDirectorySeparator(path[0]))
```

Comment:
Earlier extendedUncSyntax is set to false if extendedSyntax is false, so in what circumstance here would extendedSyntax be false but extendedUncSyntax be true?